### PR TITLE
ColumnChunk statistics for zone mapping

### PR DESCRIPTION
**Pull request by benjaminwinger**
I went a little in circles on this, trying to reduce the mess of switch cases, settling on ones using macros which I'm not overly fond of, but gets the job done at no runtime cost. I think it might be possible to create something equivalent using templates, but it's tricky.

This replaces the compression-specific data in CompressionMetadata with minimum and maximum statistics (which I considered moving out of there, but I think I prefer to keep the data separate as much as possible, since the minimum, maximum and compression type are all that's needed at the moment for the compression-related functions).

More tests are needed, including tests of tests updating and deleting nodes, and rel versions of the tests. Updates when writing don't seem to be correct as some tests are failing.
It might be useful to profile and see how much of an impact calculating the bitpacking info from the min/max has. It should be pretty fast, but also is being done for every `decompressFromPage` call now.

### CODE REVIEW

File: src/function/table_functions/call_functions.cpp [hunk 1]
```diff
@@ -481,7 +486,54 @@ void StorageInfoFunction::appendColumnChunkStorageInfo(node_group_idx_t nodeGrou
     outputChunk.getValueVector(4)->setValue<uint64_t>(vectorPos, metadata.pageIdx);
     outputChunk.getValueVector(5)->setValue<uint64_t>(vectorPos, metadata.numPages);
     outputChunk.getValueVector(6)->setValue<uint64_t>(vectorPos, metadata.numValues);
-    outputChunk.getValueVector(7)->setValue(vectorPos, metadata.compMeta.toString());
+    auto physicalType = column->getDataType()->getPhysicalType();
+    switch (physicalType) {
```
Comment (ray6080) [inline src/function/table_functions/call_functions.cpp:490]: Can we refactor this with `TypeUtils::visit` now?

**Review by ray6080 — APPROVED** [review]

File: test/test_files/update_rel/stats.test
[the remark this answers is not in the document]
Comment (benjaminwinger) [reply test/test_files/update_rel/stats.test:1]: I've added some very simple many to many tests (identical to the one to one). They are failing, along with a bunch of other rel tests; I don't think the statistics are working properly for those columns, which breaks how they are compressed and produces nonsense values.

File: src/include/storage/store/string_column.h [hunk 2]
```diff
@@ -22,9 +22,19 @@ class StringColumn final : public Column {
 
     void append(ColumnChunk* columnChunk, common::node_group_idx_t nodeGroupIdx) override;
 
-    void writeValue(const ColumnChunkMetadata& chunkMeta, common::node_group_idx_t nodeGroupIdx,
-        common::offset_t offsetInChunk, common::ValueVector* vectorToWriteFrom,
-        uint32_t posInVectorToWriteFrom) override;
+    void write(common::node_group_idx_t nodeGroupIdx, common::offset_t offsetInChunk,
+        common::ValueVector* vectorToWriteFrom, uint32_t posInVectorToWriteFrom) override;
+
+    string_index_t writeStringValue(const ColumnChunkMetadata& chunkMeta,
```
[the remark this answers is not in the document]
Comment (benjaminwinger) [reply src/include/storage/store/string_column.h]: No, it does not. I've made it protected.

File: test/test_files/update_node/stats.test [hunk 3]
```diff
@@ -0,0 +1,136 @@
+-GROUP UpdateNodeStatsTest
+-DATASET CSV empty
+
+--
+
+-CASE CreateIntegerStats
+# Note that integer compression means that values get re-compressed and are updated out of place
+-STATEMENT CREATE NODE TABLE test(id SERIAL, value INT8, PRIMARY KEY(id));
```
[the remark this answers is not in the document]
Comment (benjaminwinger) [reply test/test_files/update_node/stats.test:8]: Added, and also corrected serial commits to set and update the min/max values.

File: test/test_files/update_node/stats.test [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+---- ok
+-STATEMENT CREATE (a:test {value:1})
+---- ok
+-STATEMENT CALL storage_info('test') WHERE column_name = "value" RETURN min;
```
[the remark this answers is not in the document]
Comment (benjaminwinger) [reply test/test_files/update_node/stats.test]: Done

File: src/function/table_functions/call_functions.cpp [hunk 5]
```diff
@@ -469,7 +474,54 @@ void StorageInfoFunction::appendColumnChunkStorageInfo(node_group_idx_t nodeGrou
...
+#define CASE(physicalType, T)                                                                      \
+    case physicalType: {                                                                           \
+        auto min = metadata.compMeta.min.get<T>();                                                 \
+        auto max = metadata.compMeta.max.get<T>();                                                 \
+        outputChunk.getValueVector(7)->setValue(                                                   \
+            vectorPos, TypeUtils::castValueToString(*column->getDataType(), (uint8_t*)&min,        \
+                           outputChunk.getValueVector(7).get()));                                  \
+        outputChunk.getValueVector(8)->setValue(                                                   \
+            vectorPos, TypeUtils::castValueToString(*column->getDataType(), (uint8_t*)&max,        \
+                           outputChunk.getValueVector(7).get()));                                  \
```
[the remark this answers is not in the document]
Comment (benjaminwinger) [reply src/function/table_functions/call_functions.cpp]: Yes, I've corrected it.

**Pull request merged**